packages/react/src/ActionList/Item.tsx

```diff
@@ -259,8 +268,24 @@ export const Item = React.forwardRef<HTMLLIElement, ActionListItemProps>(
     const inlineDescriptionId = `${itemId}--inline-description`
     const blockDescriptionId = `${itemId}--block-description`
     const inactiveWarningId = inactive && !showInactiveIndicator ? `${itemId}--warning-message` : undefined
+    const validRole = listRole === 'listbox' || listRole === 'menu' || inactive
+
+    const ButtonItemWrapper = buttonSemantics
+      ? (React.forwardRef(({as: Component = 'button', children, ...props}, forwardedRef) => {
+          return (
+            <Box
+              as={Component as React.ElementType}
+              sx={merge<BetterSystemStyleObject>(styles, sxProp)}
+              ref={forwardedRef}
+              {...props}
+            >
+              {children}
+            </Box>
+          )
+        }) as PolymorphicForwardRefComponent<React.ElementType, ActionListItemProps>)
+      : React.Fragment
 
-    const ItemWrapper = _PrivateItemWrapper || React.Fragment
+    const ItemWrapper = _PrivateItemWrapper || (validRole || !buttonSemantics ? React.Fragment : ButtonItemWrapper)
```

Comment:
I think we also need to add NavList context check into this condition, don't we?


```jsx
let DefaultItemWrapper = React.Fragment 
if(buttonSemantics) {
  DefaultItemWrapper = validRole || navListContext ? React.Fragment : ButtonItemWrapper
} 

const ItemWrapper = _PrivateItemWrapper || DefautlItemWrapper
```

I don't mean to re-write the condition, it wasn't straightforward to read when I added the NavList (at least for me 😅) Do it however you feel comfortable 💖

Reply:
You're right! This is easier to read too. Thank you! I adjusted `validRole` to handle what `navListContext` would be here.